runtime/lua/vim/shared.lua

```diff
@@ -214,7 +228,7 @@ end
 ---@param t table<T, any> (table) Table
 ---@return T[] : List of keys
 function vim.tbl_keys(t)
-  vim.validate({ t = { t, 't' } })
+  expecttype(t, 'table')
```

Comment:
this could be an overload on `vim.validate`. so for the "fast path", one can call `vim.validate(val: any, type: string)` instead of the kwargs form.